core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -121,6 +122,17 @@ private[spark] class Master(
     throw new SparkException("spark.deploy.defaultCores must be positive")
   }
 
+  // Alternative application submission gateway that is stable across Spark versions
+  private val restServerEnabled = conf.getBoolean("spark.master.rest.enabled", true)
+  private val restServer =
+    if (restServerEnabled) {
+      val port = conf.getInt("spark.master.rest.port", 17077)
```

Comment:
We can use the "find free port" code to increment it if there's a collision, so I'd rather just pick a number that's easy to remember by default. Especially if this might eventually become the main protocol for app submission.